apps/api/src/routes/course/clone.ts

```diff
@@ -0,0 +1,70 @@
+import { Hono } from 'hono';
+import { ZCourseClone } from '$src/types/course';
+import { zValidator } from '@hono/zod-validator';
+import { cloneCourse } from '$src/services/course-clone';
+import { validateUser } from '$src/utils/auth/validate-user';
+import type { CloneCourseResponse, CloneCourseErrorResponse } from '$src/types/database';
+import type { User } from '@supabase/supabase-js';
+import type { TCourseClone } from '$src/types/course';
+
+export const cloneRouter = new Hono().post(
+  '/',
+  zValidator('json', ZCourseClone),
+  async (c) => {
+    try {
+      
+      const validatedData = (c.req.valid as (key: 'json') => TCourseClone)('json');
+      const { courseId, newTitle, newDescription, newSlug, organizationId } = validatedData;
+
+      // Get the authorization header
+      const authHeader = c.req.header('Authorization');
+      if (!authHeader) {
+        const errorResponse: CloneCourseErrorResponse = { error: 'Unauthorized' };
+        return c.json(errorResponse, 401);
+      }
+
+      // Extract the token
+      const token = authHeader.replace('Bearer ', '');
+
+      // Validate user and get their ID
+      let user: User;
+      try {
```

Comment:
All these auth related logic is in the `authMiddleware` in `apps/api/src/middlewares/auth.ts`.

Look at the project to see how it is used.